### PR TITLE
Checking patron group for on shelf eligibility

### LINKED ISSUE #3188: Update On Shelf Pickup Class to reflect Alma Patron Group Eligibility
The "patron_group" attribute contains the type of Alma user account group an account is part of. 

[Alma Patron Groups organized by the type of Requests Services they can utilize](https://docs.google.com/spreadsheets/d/1c4lmMiOzSqMbPYEM5Ni2pofI8lvKpv0FlaRx-xui39E/edit?gid=0#gid=0)

### Acceptance criteria 

- [x] Patron groups "P" , "REG", "GRAD", "SENR", "UGRAD" can request an on shelf pick-up. 
- [x] Patron groups "Affiliate", "Affiliate-P", "CASUAL", and "GST" cannot request an on shelf pick-up. 

## Notes
This is the [service class](https://github.com/pulibrary/orangelight/blob/main/app/models/requests/service_eligibility/on_shelf_pickup.rb) to update.

### PR DESCRIPTION
**Pull request by rladdusaw**
Resolves #3188

### CODE REVIEW

File: spec/system/searching_spec.rb [hunk 1]
```diff
@@ -194,6 +194,7 @@
     end
 
     it 'can add a facet to an existing search', js: true do
+      allow(Bibdata).to receive(:get_patron).and_return({ "patron_group" => "P" })
```
Comment (sandbergja) [inline spec/system/searching_spec.rb:197]: Why is this mock needed?  I wouldn't expect the patron group to be checked just to do an advanced search.

File: spec/features/requests/form_spec.rb [hunk 2]
```diff
@@ -24,6 +24,7 @@
   let(:valid_patron_no_campus_response) { fixture('/bibdata_patron_response_no_campus.json') }
   let(:valid_graduate_student_no_campus_response) { fixture('/bibdata_patron_response_graduate_no_campus.json') }
   let(:invalid_patron_response) { fixture('/bibdata_not_found_patron_response.json') }
+  let(:valid_patron_response_no_ldap) { fixture('/bibdata_patron_response_no_ldap.json') }
```
Comment (sandbergja) [inline spec/features/requests/form_spec.rb:27]: 👍🏻

**Review by sandbergja — COMMENTED** [review]
Looks good, thanks @rladdusaw !  I just had one question.

File: spec/system/searching_spec.rb (hunk 1 above)
Comment (christinach) [reply spec/system/searching_spec.rb:197]: @rladdusaw I have the same question

**Review by christinach — COMMENTED** [review]
@rladdusaw thank you. I have one question that is in a comment. 
Can you also rename provider_elegible? to provider_eligible?

File: spec/system/searching_spec.rb (hunk 1 above)
Comment (rladdusaw) [reply spec/system/searching_spec.rb]: That test is a little flakey and sometimes fails for me locally. I guess I got caught up in adding that line to all the failing tests toward the end. I've removed it.

**Review by christinach — APPROVED** [review]
Thanks @rladdusaw 🐈‍⬛ !

**Pull request merged**